src/components/Popup.js

```diff
@@ -20,6 +20,17 @@ class Popup {
     }
   }
 
+  renderLoading(isLoading) {
+    const submitButton = this._popup.querySelector(".modal__button");
+    if (submitButton) {
+      if (isLoading) {
+        submitButton.textContent = "Saving...";
+      } else {
+        submitButton.textContent = "Save";
```

Comment:
Please, move the `renderLoading` method into `PopupWithForm` because it's related to the submit button, but `Popup` is used for the image modal as well, which doesn't have any submit buttons